app/code/Magento/Swatches/Model/ConvertSwatchAttributeFrontendInput.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Swatches\Model;
+
+/**
+ * Performs the conversion of the frontend input value for attribute data
+ */
+class ConvertSwatchAttributeFrontendInput
+{
+    /**
+     * Performs the conversion of the frontend input value for attribute data
+     *
+     * @param array|null $data
+     *
+     * @return array|null
+     */
+    public function execute(?array $data): ?array
```

Comment:
I was thinking about the implementation for some time. Must admit that the service should not receive null as the argument since it breaks the implementation consistency. We shall try to be as more specific as possible in the types of accepting/returning in order to make the logic transparent. 
From my point of view, if don't have the `$data`, we should not call the service. On the other hand, the service should know what's the structure of the data and then make the decision regarding the conversion process. So, let's leave it as it is.

Thank you

Reply:
Thanks for your thoughts. Just to clarify, in this case this part of the logic was copied from existing logic with unit test which covers the case with NULL for some reason, that's why I decided to leave it as it was.